docs/resources/routing_rules.md

```diff
@@ -143,3 +114,35 @@ Optional:
 - `key` (String) The value being filtered. Required if the filter effect is 'keep' or 'remove'.
 See [docs](https://docs.p0.dev/just-in-time-access/request-routing#resource) for available values.
 - `pattern` (String) Filter patterns. Patterns are unanchored.
+
+
+
+<a id="nestedblock--rule--approval"></a>
+### Nested Schema for `rule.approval`
+
+Required:
+
+- `type` (String) Determines trust requirements for access. If empty, access is disallowed. Except for 'deny', meeting any requirement is sufficient to grant access. Possible values:
+    - 'auto': Access is granted according to the requirements of the specified 'integration'
+    - 'deny': Access is always denied
+    - 'escalation': Access may be approved by on-call members of the specified services, who are paged when access is requested
+    - 'group': Access may be granted by any member of the defined directory group
+    - 'p0': Access may be granted by any user with the P0 approval role (defined in the P0 app)
+
+Optional:
+
+- `directory` (String) May only be used if 'type' is 'group'. One of "azure-ad", "okta", or "workspace".
+- `id` (String) May only be used if 'type' is 'group'. This is the directory's internal group identifier for matching approvers.
+- `integration` (String) May only be used if 'type' is 'auto' or 'escalation'. Possible values:
+    - 'pagerduty': Access is granted if the requestor is on-call.
+- `label` (String) May only be used if 'type' is 'group'. This is any human-readable name for the directory group specified in the 'id' attribute.
+- `options` (Attributes) If present, determines additional trust requirements. (see [below for nested schema](#nestedatt--rule--approval--options))
+- `services` (List of String) May only be used if 'type' is 'escalation'. Defines which services to page on escalation.
```

Comment:
I would reorder them so type=group and type=auto/escalation bullet points are together
```suggestion
- `directory` (String) May only be used if 'type' is 'group'. One of "azure-ad", "okta", or "workspace".
- `id` (String) May only be used if 'type' is 'group'. This is the directory's internal group identifier for matching approvers.
- `label` (String) May only be used if 'type' is 'group'. This is any human-readable name for the directory group specified in the 'id' attribute.
- `integration` (String) May only be used if 'type' is 'auto' or 'escalation'. Possible values:
    - 'pagerduty': Access is granted if the requestor is on-call.
- `services` (List of String) May only be used if 'type' is 'escalation'. Defines which services to page on escalation.
- `options` (Attributes) If present, determines additional trust requirements. (see [below for nested schema](#nestedatt--rule--approval--options))
```

Reply:
This is just a code move; maybe we can address these markdown suggestions in a separate PR?

Reply:
I prefer alphabetical ordering as it's more stable

Reply:
This is automatically generated and will always be alphabetical.